frontend-wasm/src/miden_abi/mod.rs

```diff
@@ -0,0 +1,42 @@
+pub(crate) mod transform;
+pub(crate) mod tx_kernel;
+
+use miden_core::crypto::hash::RpoDigest;
+use miden_hir::Symbol;
+use miden_hir_type::MidenAbiFunctionType;
+
+/// Parse the stable import function name and the hex encoded digest from the function name
+pub fn parse_import_function_digest(import_name: &str) -> Result<(String, RpoDigest), String> {
+    // parse the hex encoded digest from the function name in the angle brackets
+    // and the function name (before the angle brackets) example:
+    // "miden:tx_kernel/note.get_inputs"
+    let mut parts = import_name.split('<');
+    let function_name = parts.next().unwrap();
+    let digest = parts
+        .next()
+        .and_then(|s| s.strip_suffix('>'))
+        .ok_or_else(|| "Import name parsing error: missing closing angle bracket in import name")?;
+    Ok((
+        function_name.to_string(),
+        RpoDigest::try_from(digest).map_err(|e| e.to_string())?,
+    ))
+}
+
+pub fn is_miden_sdk_module(module_id: Symbol) -> bool {
+    tx_kernel::types().contains_key(module_id.as_str())
+}
+
+/// Get the target Miden ABI tx kernel function type for the given module and function id
+#[inline(always)]
```

Comment:
Probably not worth inlining this call, since all of the panic infra will be here, there's a good chance LLVM won't even _try_ to inline it. Even if it does though, it will reduce the opportunity to inline more important things.

If you think it's worth it though, I'd maybe just change it to `#[inline]`, to reflect that it's just a hint, rather than trying to force it. Ultimately, both rustc and LLVM are free to ignore either one, but that could change at any time in the future.

Reply:
Good catch! Thanks. I have no reason to inline it. I probably added it automatically after spending some time in the bindings crate, where almost everything is unlined. Fixed.

Reply:
Yeah, I use `#[inline]` and `#[inline(always)]` _a lot_, mostly because we're frequently working on pretty low-level stuff where it can make a noticeable benefit. A lot of people don't realize that without those annotations, Rust won't do much inlining at all on its own, instead it relies entirely on LLVM to figure things out.

That said, I think there are probably quite a few places where I've used them inappropriately, so I'm probably not always setting the best example here 😅